core/chains/evm/txmgr/txmgr.go

```diff
@@ -248,20 +279,45 @@ func (b *Txm) runLoop(eb *EthBroadcaster, ec *EthConfirmer) {
 			eb.Trigger(address)
 		case head := <-b.chHeads:
 			ec.mb.Deliver(head)
+		case f := <-b.reset:
+			if err := eb.Close(); err != nil {
+				panic(err)
+			}
+			if err := ec.Close(); err != nil {
+				panic(err)
+			}
+
+			f()
+
+			eb = NewEthBroadcaster(b.db, b.ethClient, b.config, b.keyStore, b.eventBroadcaster, keyStates, b.gasEstimator, b.resumeCallback, b.logger, b.checkerFactory)
+			ec = NewEthConfirmer(b.db, b.ethClient, b.config, b.keyStore, keyStates, b.gasEstimator, b.resumeCallback, b.logger)
+
+			// FIXME: the nonce syncer will run here, we probably don't want that
+			if err := eb.Start(ctx); err != nil {
+				b.logger.Criticalw("Failed to start EthBroadcaster", "error", err)
```

Comment:
Should these panic too?
```suggestion
				b.logger.Panicw("Failed to start EthBroadcaster", "error", err)
```

Reply:
Start is more complex than Close, I am not convinced it's impossible for it to fail to start. We may need to put it on a retry loop instead?